x-pack/plugins/infra/server/routes/overview/lib/create_top_nodes_query.ts

```diff
@@ -13,10 +13,41 @@ export const createTopNodesQuery = (
   options: TopNodesRequest,
   source: MetricsSourceConfiguration
 ) => {
+  const nestedSearchFields: { [key: string]: string } = {
+    rx: 'rx>bytes',
+    tx: 'tx>bytes',
+  };
   const sortByHost = options.sort && options.sort === 'name';
-  const sortField = sortByHost ? '_key' : options.sort ?? 'uptime';
+  const metricsSortField = options.sort
+    ? nestedSearchFields[options.sort] || options.sort
+    : 'uptime';
+  const sortField = sortByHost ? '_key' : metricsSortField;
   const sortDirection = options.sortDirection ?? 'asc';
   return {
+    runtime_mappings: {
+      rx_bytes_per_period: {
+        type: 'double',
+        script: {
+          source: `
+          if(doc[\'host.network.ingress.bytes\'].size() !=0)
```

Comment:
We would want to calculate the field when its zero, just not when it doesn't exist on a document.  When it doesn't exist it means we could be getting the metricset.period of the wrong metricset so we need to make sure to filter those docs out or the rate would be incorrect.  

We should probably create a test for this if one doesn't exist.  We could test some archive data with a small amount of documents with a few different metricsets and metricset periods and make sure it calculates as expected.

Reply:
@mohamedhamed-ahmed explained to me that this is correct because size() only returns 1 or 0 depending if the field exists.